ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java

```diff
@@ -204,14 +208,17 @@ private void addAttributesFromPayload(Map<String, String> attributes, ObjectNode
           .ifPresent(v -> attributes.put(Attribute.OS_VERSION, v));
     }
 
-    // Try extracting variants of top-level client id.
-    Stream
-        .of(attributes.get(Attribute.CLIENT_ID), json.path(Attribute.CLIENT_ID).textValue(),
-            json.path("clientId").textValue())
-        .map(ParsePayload::normalizeUuid) //
-        .filter(Objects::nonNull) //
-        .findFirst() //
-        .ifPresent(v -> attributes.put(Attribute.CLIENT_ID, v));
+    if (Objects.isNull(attributes.get(Attribute.CLIENT_ID))) {
```

Comment:
```suggestion
    if (attributes.get(Attribute.CLIENT_ID) == null) {
```

Is there a reason to use Objects.isNull here vs. the simpler `== null`? I think `isNull` and `nonNull` are primarily useful because they can be used as method references in Stream computations.

I'd also replace the `Objects.nonNull` call three lines down with `!= null` since it's no longer a simple method reference.